### DISCUSSION TITLE
Taskbar icon appears for GUI apps, but no window

### DISCUSSION THREAD

**Discussion by usernamenumber** [microsoft/wslg#370] [Q&A]
- Windows version: 21H2 build 22000.71   
  - 👆 This is Windows 11 Pro, but from the discussion I've seen here, 21H2 is supposed to support wslg? 
- Diagnostic info, as requested on the wiki:

    ```
    $ cat /mnt/wslg/versions.txt
    WSLg ( x86_64 ): 1.0.19+Branch.main.Sha.2a47bf20c239b523d7413f315cbc9ee60dfde831
    Mariner: VERSION="1.0.20210224"
    FreeRDP: 5f083fa0b97d433d6204985f6047886e29c1c61e
    weston: 16de531f00aa3dfd17e0de74c8f49e9fd7cec617
    pulseaudio: 2f0f0b8c3872780f15e275fc12899f4564f01bd5
    mesa:
    
    $ echo $DISPLAY
    :0                                                                                                                             
    
    $ ls -la /tmp/.X11-unix
    lrwxrwxrwx 1 root root 19 Jul 16 11:21 /tmp/.X11-unix -> /mnt/wslg/.X11-unix
    
    $ ls -la /tmp/.X11-unix/
    total 0
    drwxrwxrwx 2 root root   60 Jul 16 11:21 .
    drwxrwxrwt 5 root root  220 Jul 18 10:02 ..
    srwxrwxrwx 1 brad users   0 Jul 16 11:21 X0        
    ```

---

When I try running a GUI app from WSL2 (in these screenshots it's firefox, but I've tried xeyes and others with the same results), a taskbar icon is created (in this case a firefox logo with the Tux overlay - cute!), but no window appears.

I do see slightly different empty boxes If I mouseover the taskbar icon 

![image](https://user-images.githubusercontent.com/188349/126071420-a8e8051c-4b81-488e-b651-c453b45513b3.png)

or alt-tab

![wsl-ff](https://user-images.githubusercontent.com/188349/126071198-dadaee32-38c5-49a4-94d1-d5013b5cd8eb.png)

but it never gets past that point. 

One other odd behavior that may or may not be a clue: when I go back to the WSL window, `^c` doesn't do anything until I press Enter, and then the command responds to `^c` and exits. 

I have tried replacing my .bashrc with the stock file from `/etc/skel`, but no change.

**Comment by onomatopellan** [comment]
From an elevated cmd run `wsl.exe --shutdown` and after that `wsl.exe --update` and try again.

**Reply by usernamenumber** [reply]
That did it. Thanks!

**Reply by colin0117** [reply]
That does resolve it for me too, but it's happening fairly often and needing that restart. I'm on W11 (10.0.22000 Build 22000), everything updated and with a new Ubuntu 22.04 distro. When it first comes up, all UIs appear as expected, but after a while, I'm seeing the issue reported in the OP.

**Reply by onomatopellan** [reply]
@colin0117 WSLg has changed its way of being updated. Now to get latest version with [tons of fixes](https://github.com/microsoft/WSL/releases) you just need to make sure you are installing latest WSL from the Store https://aka.ms/wslstorepage
